neutrino.go

```diff
@@ -809,7 +827,17 @@ func NewChainService(cfg Config) (*ChainService, error) {
 		GetBlock:         s.GetBlock,
 		firstPeerSignal:  s.firstPeerConnect,
 		queryAllPeers:    s.queryAllPeers,
-	})
+	}
+
+	if cfg.BlkHdrSideLoad != nil {
+		blkMgrConfig.blkHdrSideLoad = &SideLoadOpt{
+			SkipVerify: cfg.BlkHdrSideLoad.SkipVerify,
+			Reader:     cfg.BlkHdrSideLoad.Reader,
+			SourceType: cfg.BlkHdrSideLoad.SourceType,
+		}
+	}
+
+	bm, err := newBlockManager(blkMgrConfig)
```

Comment:
Rather than make this the responsibility of the block manager, we can load the header right before we start up the block manager.